File: CPP/20_Longest_Common_Prefix.cpp
/*
Write a function to find the longest common prefix string amongst an array of strings.

If there is no common prefix, return an empty string "".
*/

string longestCommonPrefix(vector<string>& strs) {
    sort(strs.begin(), strs.end());
    string common_prefix = "";
    for(int i = 0; i < min(strs[0].size(), strs[strs.size() - 1].size()); i++){
        if(strs[0][i] != strs[strs.size() - 1][i]) {
            return common_prefix;
        }
        common_prefix += strs[0][i];
    }
    return common_prefix;
}

File: Python/20_Longest_Common_Prefix.py
'''
Write a function to find the longest common prefix string amongst an array of strings.

If there is no common prefix, return an empty string "".
'''

def longestCommonPrefix(strs):
    strs = sorted(strs)
    common_prefix = ""
    for i in range(min(len(strs[0]), len(strs[-1]))):
        if strs[0][i] != strs[-1][i]:
            return common_prefix
        common_prefix += strs[0][i]
    return common_prefix